client.go

```diff
@@ -0,0 +1,78 @@
+package filetypes
+
+import (
+	"github.com/cloudquery/filetypes/csv"
+	"github.com/cloudquery/filetypes/json"
+)
+
+type FormatType string
+
+const (
+	FormatTypeCSV  = "csv"
+	FormatTypeJSON = "json"
+)
+
+type FileSpec struct {
+	Format         FormatType `json:"format,omitempty"`
+	IncludeHeaders bool       `json:"include_headers,omitempty"`
+	Delimiter      rune       `json:"delimiter,omitempty"`
+}
+
+func (s *FileSpec) SetDefaults() {
+	if s.Delimiter == 0 {
+		s.Delimiter = ','
+	}
+}
+
+type Client struct {
+	spec                   *FileSpec
+	csv                    *csv.Client
```

Comment:
can client, transformer + reverse transformer be grouped into another struct?
this way you'll have
```go
type Client struct {
	spec *FileSpec
	csv *wrapper // example name
	json *wrapper
}
```

Reply:
This code is not duplicated anywhere, so we would just be adding layers of abstraction/indirection without seeing any reduction in LOC or complexity